kubernetes/auth.go

```diff
@@ -0,0 +1,36 @@
+package kubernetes
+
+import (
+	"log"
+	"os"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Config contains the kubernetes clientset and configs.
+type Config struct {
+	clientset *kubernetes.Clientset
+}
+
+// authenticate authenticates and authorizes the client.
+func (k *Config) authenticate() (err error) {
+	var config *rest.Config
+	kubeconfigPath := os.Getenv("KUBECONFIG")
+	if len(kubeconfigPath) > 0 {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		log.Printf("Cannot connect to Kubernetes: %v\n", err)
```

Comment:
one small suggestion: either use `https://github.com/TRON-US/go-common/blob/master/db/postgres/hooks.go#L6` to do consistent logging with the rest of the modules, or return this message as an error directly and let upper level handle logging

Reply:
Done!